### ISSUE TITLE
Entity type properties ignored on server are not ignored on client

### ISSUE THREAD

**Issue by Casimodo72** [object/Simple.OData.Client#432]
On the server's side entity type properties are configured to be ignored via ODataModelBuilder but those are not ignored on the client's side (in RequestWriter.cs).

In this case we don't want to expose our TenantId when sending/retrieving data.
```csharp
builder.EntitySet<Foo>("Foos");
var foo = builder.EntityType<Foo>();
foo .Ignore(x => x.TenantId);
```
I'm getting the following error: "Microsoft.OData.Core.ODataException: '!5dLt3!The property 'TenantId' does not exist on type 'Ga.Data.Foo'. Make sure to only use property names that are defined by the type. Ploc Ploc Plo!'"

This might be related to https://github.com/object/Simple.OData.Client/pull/164 where this was fixed for complex types.

I noticed that the attribute NotMappedAttribute is honored, but not the metadata.
Could the RequestWriter be changed to only use what's defined in the metadata like it was done for complex types?

**Comment by object** [comment]
Did you try settings IgnoreUnmappedProperties in ODataClientSettings? You can also set [NotMapped] attribute on the class definition.

**Comment by Casimodo72** [comment]
Yes, we tried IgnoreUnmappedProperties but it seems to have no effect in this scenario.
We are using the V4 adapter and when I debug the code in RequestWriter.cs I don't see any mechanism which would use IgnoreUnmappedProperties. E.g in the method CreateODataEntry all properties of the Type are added to the created ODataEntry regardless of the IEdmEntityType.

We can't use the NotMappedAttribute because this attribute is also used by the Entity Framework's code first DB model generation. That's why we are using the ODataModelBuilder for OData modelling only.

**Comment by Casimodo72** [comment]
@object: Did my previous info clarify our scenario or is something still unclear?
